### ISSUE TITLE
Data Dumps not auto-generated

### ISSUE THREAD

**Issue by neilt** [internetarchive/openlibrary#9521]
### Problem

As of today July 3, https://archive.org/details/ol_exports?sort=-publicdate does not show a June 2024 data dump.  

And https://openlibrary.org/data/ol_dump_latest.txt.gz is still downloading  `ol_dump_2024-05-31.txt.gz`

The dumps usually generate by the 1st or 2nd day of the next month.

### 2024-12 Update

The problem is occurring again and @mekarpeles is investigating

**Comment by tfmorris** [comment]
The dump program was recently changed, so this could be related. https://github.com/internetarchive/openlibrary/pull/9127

**Comment by mekarpeles** [comment]
While running through our diagnosing cron failures guide (https://github.com/internetarchive/olsystem/wiki/Crons#diagnosing-cron-failures) we discovered:

```
DEBUG    : stats.py    :  46 :  Postgres Database : coverstore
Exception ignored in atexit callback: <function AtexitIntegration.setup_once.<locals>._shutdown at 0x7fa304df2660>
Traceback (most recent call last):
  File "/home/openlibrary/.local/lib/python3.12/site-packages/sentry_sdk/integrations/atexit.py", line 61, in _shutdown
    client.close(callback=integration.callback)
  File "/home/openlibrary/.local/lib/python3.12/site-packages/sentry_sdk/client.py", line 580, in close
    self.flush(timeout=timeout, callback=callback)
  File "/home/openlibrary/.local/lib/python3.12/site-packages/sentry_sdk/client.py", line 604, in flush
    self.transport.flush(timeout=timeout, callback=callback)
  File "/home/openlibrary/.local/lib/python3.12/site-packages/sentry_sdk/transport.py", line 525, in flush
    self._worker.submit(lambda: self._flush_client_reports(force=True))
  File "/home/openlibrary/.local/lib/python3.12/site-packages/sentry_sdk/worker.py", line 117, in submit
    self._ensure_thread()
  File "/home/openlibrary/.local/lib/python3.12/site-packages/sentry_sdk/worker.py", line 42, in _ensure_thread
    self.start()
  File "/home/openlibrary/.local/lib/python3.12/site-packages/sentry_sdk/worker.py", line 70, in start
    self._thread.start()
  File "/home/openlibrary/.local/lib/python3.12/site-packages/sentry_sdk/integrations/threading.py", line 56, in sentry_start
    return old_start(self, *a, **kw)
           ^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/local/lib/python3.12/threading.py", line 992, in start
    _start_new_thread(self._bootstrap, ())
RuntimeError: can't create new thread at interpreter shutdown
```

**Comment by cdrini** [comment]
This was also added which might cause the error: https://github.com/internetarchive/openlibrary/pull/9369

**Comment by mekarpeles** [comment]
@mekarpeles to manually run data dumps script according to
https://github.com/internetarchive/olsystem/wiki/Crons#monthly-data-dumps

**Comment by mekarpeles** [comment]
The problem was a missing `fi` https://github.com/internetarchive/openlibrary/blob/master/scripts/oldump.sh#L125-L126

**Comment by mekarpeles** [comment]
We manually ran for 2024-12 and everything worked but we need to look back at more logs and also verify things work for 2025-01 before we close.

**Comment by mekarpeles** [comment]
Our logs reveal dumps failed from cron container because the container did not correctly mount `/olsystem` and thus configs and scripts are unavailable to it:

```
+ log '=== Step 7 ==='
++ date '+%Y-%m-%d %H:%M:%S'
+ echo '2025-01-01 00:01:25 [openlibrary.dump] * === Step 7 ==='
2025-01-01 00:01:25 [openlibrary.dump] * === Step 7 ===
++ compgen -G 'ol_dump_*_2024-12*.txt.gz'
+ [[ ! -f ol_dump_ratings_2024-12-31.txt.gz
ol_dump_covers_metadata_2024-12-31.txt.gz
ol_dump_reading-log_2024-12-31.txt.gz ]]
+ mkdir -p /1/var/tmp/oldumpsort
+ log 'splitting the dump: ol_dump_%s_2024-12-31.txt.gz -- takes approx. 85 minutes for 68,000,000+ records...'
++ date '+%Y-%m-%d %H:%M:%S'
+ echo '2025-01-01 00:01:25 [openlibrary.dump] * splitting the dump: ol_dump_%s_2024-12-31.txt.gz -- takes approx. 85 minutes for 68,000,000+ records...'
2025-01-01 00:01:25 [openlibrary.dump] * splitting the dump: ol_dump_%s_2024-12-31.txt.gz -- takes approx. 85 minutes for 68,000,000+ records...
+ gzip -cd ol_dump_2024-12-31.txt.gz
+ python /openlibrary/scripts/oldump.py split --format ol_dump_%s_2024-12-31.txt.gz
Couldn't find statsd_server section in config
2025-01-01 00:01:27 [openlibrary.dump] ['/openlibrary/scripts/oldump.py', 'split', '--format', 'ol_dump_%s_2024-12-31.txt.gz'] on Python 3.12.2
Traceback (most recent call last):
  File "/openlibrary/scripts/oldump.py", line 37, in <module>
    load_config(ol_config)
  File "/openlibrary/openlibrary/config.py", line 41, in load_config
    infogami.load_config(config_file)
  File "/openlibrary/infogami/__init__.py", line 188, in load_config
    with open(config_file) as in_file:
         ^^^^^^^^^^^^^^^^^
FileNotFoundError: [Errno 2] No such file or directory: '/olsystem/etc/openlibrary.yml'
```

**Comment by mekarpeles** [comment]
@cdrini recently fixed olsystem on this machine so the cron container (and other containers) have it. 

I am going to patch-deploy an update to the cron that runs a db dump **once** today (and only once) and this will disappear with our next deploy.

So let's hold off on deploys for a bit, which shouldn't be hard given #10255

**Comment by mekarpeles** [comment]
Ran successfully

**Comment by jimchamp** [comment]
Looks like this failed during [step 7](https://github.com/internetarchive/openlibrary/blob/c62909e6571e5e81153dfa67bdaf7b1f76c68011/scripts/oldump.sh#L176) of the process.

The dumps do appear to exist in the cron job container at `/1/var/tmp/dumps`, but the `oldumpsort` directory was not [removed](https://github.com/internetarchive/openlibrary/blob/c62909e6571e5e81153dfa67bdaf7b1f76c68011/scripts/oldump.sh#L182).  Perhaps we can manually upload the dumps and do any necessary clean-up?

Also noting a number of `UnboundLocalError`s are raised during some steps of the process.  These occur in logging statement that try to access local variable `i` outside of the variable's scope.  I _do not_ believe that these making our dumps fail, and will open a PR that addresses these shortly.

**Comment by jimchamp** [comment]
While dump files do exist in the container, the `ol_cdump` file and its derivatives are quite small.  It looks like an issue occurred during the `cdump` generation ([step 5](https://github.com/internetarchive/openlibrary/blob/f5e32ce97c1ea6d1174ea4ddafa976d04519357d/scripts/oldump.sh#L152)).

```
2025-02-01 03:28:44 [openlibrary.dump] generate_cdump(data.txt.gz, 2025-01-31) reading
2025-02-01 03:28:44 [openlibrary.dump] read_data_file(data.txt.gz, max_lines=all)
Traceback (most recent call last):
  File "/openlibrary/scripts/oldump.py", line 43, in <module>
    dump.main(sys.argv[1], sys.argv[2:])
  File "/openlibrary/openlibrary/data/dump.py", line 362, in main
    func(*args, **kwargs)
  File "/openlibrary/openlibrary/data/dump.py", line 134, in generate_cdump
    print_dump(read_data_file(data_file, max_lines), filter=filter)
  File "/openlibrary/openlibrary/data/dump.py", line 39, in print_dump
    for i, raw_json_data in enumerate(json_records):
  File "/openlibrary/openlibrary/data/dump.py", line 82, in read_data_file
    for i, line in enumerate(xopen(filename, "rt")):
                   ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
ValueError: I/O operation on closed file.
```

**Comment by jimchamp** [comment]
This should probably remain open until the dumps are generated and published.

**Comment by tfmorris** [comment]
The [2025-02-11 dump](https://archive.org/details/ol_dump_2025-02-11) is still incomplete, missing both editions, ratings, lists and other subsets. The last update date is Feb. 14, so I don't think it's still being uploaded.

There is also no cdump version for this month, so that can't be used as fallback to derive the missing datasets.

**Comment by mekarpeles** [comment]
Manually ran for 2025-03-01... Likely failed due to existing 2025-02 files still on disk ☹

**Comment by mekarpeles** [comment]
Data dumps completed successfully, we can re-open when inevitably there's another issue in the future. :)

**Comment by neilt** [comment]
Yep, just noticed this.  Thanks for re-opening.

**Comment by jimchamp** [comment]
We're going to re-run the dumps today.  If successful, they should be available within two days.

**Comment by jimchamp** [comment]
Dumps were saved to the Archive on 8 November.

**Comment by tfmorris** [comment]
Thanks for reopening. It looks like the 30 Nov 2025 dump currently consists of just redirects, authors, and works, all created on 2 Dec, which is a little weird because I thought the subsets were created from the full dump, which doesn't seem to be present.

**Comment by jimchamp** [comment]
Yeah, this was a little weird.  I'm noticing that _all_ of the dumps were generated and still exist on the server, so we shouldn't have to re-run the entire process.  Will try to get these uploaded later today.

**Comment by jimchamp** [comment]
The individual data dumps for November have been updated, and are available [here](https://archive.org/details/ol_dump_2025-11-30).

November's complete dump is being uploaded now.

**EDIT:** All of November's data dumps have been uploaded.

**Comment by tfmorris** [comment]
Thanks! 17 1/2 minutes to download everything except the full dump (136 Mbit/sec). That must be a new record! Historically it's taken several times that amount of time.
